include/oneapi/dpl/experimental/kt/esimd_radix_sort.h

```diff
@@ -81,7 +82,91 @@ radix_sort_by_key(sycl::queue __q, _KeysIterator __keys_first, _KeysIterator __k
     auto __vals_keep = oneapi::dpl::__ranges::__get_sycl_range<sycl::access_mode::read_write, _ValsIterator>();
     auto __vals_rng = __vals_keep(__vals_first, __vals_first + (__keys_last - __keys_first)).all_view();
     auto __pack = __impl::__rng_pack{::std::move(__keys_rng), ::std::move(__vals_rng)};
-    return __impl::__radix_sort<__is_ascending, __radix_bits>(__q, ::std::move(__pack), __param);
+    return __impl::__radix_sort<__is_ascending, __radix_bits, /*__in_place=*/true>(__q, __pack, __pack, __param);
+}
+
+template <bool __is_ascending = true, ::std::uint8_t __radix_bits = 8, typename _KernelParam, typename _KeysRng1,
+          typename _KeysRng2>
+std::enable_if_t<!oneapi::dpl::__internal::__is_iterator_type_v<_KeysRng1>, sycl::event>
+radix_sort(sycl::queue __q, _KeysRng1&& __keys_rng, _KeysRng2&& __keys_rng_out, _KernelParam __param = {})
+{
+    __impl::__check_esimd_sort_params<__radix_bits, _KernelParam::data_per_workitem, _KernelParam::workgroup_size>();
+    if (__keys_rng.size() == 0)
```

Comment:
Let's change the check to `... == 1` here and in other places. We do not want to sort 1 element.

Reply:
If we do this, then we need to have a separate mechanism to copy 1 element from the input to the output.  
I suppose we can do this as an optimization, but I was hoping to avoid having to specially handle such a case for kernel templates.  (We cant just noop because the output needs to be populated).
While `n` may be a runtime parameter, I think that a user who is calling kernel templates has more responsibility to avoid doing something like sorting a set of `1` element on the device.  For me, within KT there is less responsibility to provide special casing like this on behalf of the user.

Reply:
I agree with you. Thanks for describing one more subtle (at least for me) detail. 

What about removing the assert below?
 https://github.com/oneapi-src/oneDPL/blob/f371fb91935e9bc1076f7881853d200f894a7bbc/include/oneapi/dpl/experimental/kt/internal/esimd_radix_sort_dispatchers.h#L310

Reply:
Aha! good catch.  Yes I'll make the change.

Reply:
I adjusted this to `__n > 0` because we still enforce that.